### PR TITLE
List all generated files

### PR DESCRIPTION
**Pull request by arielshaqed**
Partially in order to see what files exist -- but also as a manual test.

### CODE REVIEW

File: test/spark/app/src/main/scala/Sonnets.scala [hunk 1]
```diff
@@ -41,6 +62,27 @@ object Sonnets {
         )
         target.save(outputPath)
 
+        {
```
Comment (nopcoder) [inline test/spark/app/src/main/scala/Sonnets.scala:65]: question: any reason why it needs to be scoped?

File: test/spark/app/src/main/scala/Sonnets.scala [hunk 2]
```diff
@@ -41,6 +62,27 @@ object Sonnets {
...
+           */
+          val pattern: Regex = s"${Regex.quote(outputPath)}/partition_key=[^/]*/.*/([^/]*\\.${Regex.quote(fmt)}|_SUCCESS)".r
+          val path = new Path(outputPath)
+          val fs = path.getFileSystem(sc.hadoopConfiguration)
+          val badFiles = convertToScalaIterator(fs.listFiles(path, true)).
+            map(file => s"${file.getPath.getParent}/${file.getPath.getName}").
+            filter(name => name match {
+              case pattern() => false
+              case _ => true
+            })
```
Comment (nopcoder) [inline test/spark/app/src/main/scala/Sonnets.scala:79]: I think you can keep pattern as string and just use something like 'filter(name => name match pattern)

**Review by nopcoder — APPROVED** [review]
Approved - just question and a small suggestion

File: test/spark/app/src/main/scala/Sonnets.scala (hunk 1 above)
Comment (arielshaqed) [reply test/spark/app/src/main/scala/Sonnets.scala:65]: Just limiting the scope of the variables.

**Review by arielshaqed — COMMENTED** [review]
Thanks!

Pulling -- I think the added test coverage is worth it.

**Pull request merged**